src/test/deploy/functions/validate.spec.ts

```diff
@@ -12,8 +12,8 @@ const cjson = require("cjson");
 describe("validate", () => {
   describe("functionsDirectoryExists", () => {
     const sandbox: sinon.SinonSandbox = sinon.createSandbox();
-    let resolvePpathStub: sinon.SinonStub;
-    let dirExistsStub: sinon.SinonStub;
+    let resolvePpathStub: sinon.SinonStub<[string, string], string>;
```

Comment:
What's the pattern here? Is the first arg the args to the stubbed function and the second arg is the return type?

Reply:
it's `<argument_types_in_an_array, return_type>`.

I'm not sure how I feel about it. I think the default of Sinon was much nicer before, and now it's gone all weird. I think if we _don't_ define them up front like this (and instead to `stub = sinon.Stub(thing, 'method')`), we don't run into this as often

Reply:
Agree this reads kinda weird but if that's what Sinon wants I guess it's fine. I do see a lot of use cases for defining stubs without initializing though, as we initialize them in the `beforeEach` methods and the variable needs visibility for other tests.